R/utils.R

```diff
@@ -6,7 +6,12 @@ is_pipe <- function(x) {
 }
 
 .infixes_assign <- c("<-", "<<-", "->", "->>", "=")
-.infixes <- c("+", "-", "*", "/", "^", "%%", "%/%", "%in%", .infixes_assign)
+.infixes_comp <- c("==", "!=", ">", ">=", "<", "<=")
+.infixes <- c(
+  "+", "-", "*", "/", "^", "$", "[", "[[", "!", "%%", "%/%", "%in%", 
+  .infixes_assign,
+  .infixes_comp
+)
 
 is_infix <- function(x, infix_vals = .infixes) {
 
```

Comment:
Should this function be updated to something like "If function doesn't start with an alpha char, and is a call, then it is an infix op", rather than checking against a white list of functions.

Reply:
Yeah this function feels particularly brittle, especially considering the change I'm making here. Also: we should consider other infix operators defined by other packages, e.g. `%AND%` or `%+%` or other random things.

On the other hand, what about bizarre function names?

```
str2lang("`12`(13)")[[1]]
#> `12`
as.character(str2lang("`12`(13)")[[1]])
#> [1] "12"
```

I'm going to add the change here for now and we can come back to this in a separate PR